File: CPP/003_Largest prime factor.cpp
#include <iostream>
#include <cmath>

using namespace std;

 bool isprime(long long n)
        {
                int f=2;
            bool flag=true;
            while(f <= sqrt(n))
            {
                if(n%f==0)
                {
                    flag = false;
                    break;
                }
                f++;
                
            }
            return flag;
        }
 


int main()
{
    long long num = 600851475143ll;
                    long long pf = num;
                    long long factor = 2ll;
                   
                    
                    while (factor < pf)
                    {
                        if(isprime(factor))
                        {
                            if(pf%factor==0ll)
                            {
                                pf = pf / factor;
                                
                                if(isprime(pf)==1)
                                    break;
                            }
                            else
                                factor++;
                            
                        }
                        else
                        factor++;
                    }
                    cout << "largest prime factor is " << pf << endl;
    return 0;
}

File: JAVA/003_Largest prime factor.java
class Version1 {

	
	public static void main(String[] args) 
	{
		 long num = 600851475143l;
		    long pf = num;
		    long factor = 2l;
		   
		    
		    while (factor < pf)
		    {
		        if(isprime(factor)==1)
		        {
		            if(pf%factor==0l)
		            {
		                pf = pf / factor;
		                
		                if(isprime(pf)==1)
		                    break;
		            }
		            else
		                factor++;
		            
		        }
		        else
		        factor++;
		    }
		    
		    System.out.println(" largest prime factor is "+pf);

	}
	
	public static int isprime(long n)
	{
		int f=2;
	    int flag=1;
	    while(f <= (n/2))
	    {
	        if(n%f==0)
	        {
	            flag = 0;
	            break;
	        }
	        f++;
	        
	    }
	    return flag;
	}

}
